helix-view/src/document.rs

```diff
@@ -982,6 +990,21 @@ impl Document {
         server.is_initialized().then(|| server)
     }
 
+    pub fn differ(&self) -> Option<&Differ> {
+        self.differ.as_ref()
+    }
+
+    /// Intialize/updates the differ for this document with a new base.
+    pub fn set_diff_base(&mut self, diff_base: Vec<u8>) {
+        if let Ok((diff_base, _)) = from_reader(&mut Cursor::new(diff_base), Some(self.encoding)) {
```

Comment:
`Cursor::new` might be unnecessary since [slice implements `Read`.](https://doc.rust-lang.org/std/primitive.slice.html#impl-Read)

Reply:
I somehow taught that the default `Read` implementation reads the slice in reverse and thats why I used the cursor.
That makes no sense actually so not sure where that taught came from.
I have removed the `Cursor::new` call.